### ISSUE TITLE
TFJob operator v1alpha2 doesn't work with TF.Estimator API for TF <=1.6

### ISSUE THREAD

**Issue by gaoning777** [kubeflow/kubeflow#1283]
Hi,
**Background**: converting my tfjob samples to support kubeflow v1alpha2.
**Problem**: the job waits forever, logs being "tensorflow:Waiting for model to be ready.  Ready_for_local_init_op:  Variables not initialized:...."
**The tfjob yaml:**
_apiVersion: kubeflow.org/v1alpha2
kind: TFJob
metadata:
  name: trainer-tfjob
spec:
  tfReplicaSpecs:
    MASTER:
      replicas: 1
      restartPolicy: OnFailure
      template:
        spec:
          containers:
          - command:
            - python
            - -m
            - trainer.task
            - --optimizer=Adagrad
            - --target=label
            - --learning-rate=0.1
            - --job-dir=gs://ngao-test-mlpipeline-userguide/flowermodel/kubeflow-training-974xq/train
            - --hidden-layer-size=100,50
            - --steps=2000
            - --transformed-data-dir=gs://ngao-test-mlpipeline-userguide/flowermodel/kubeflow-training-974xq/transformed
            - --schema=gs://ml-pipeline-playground/flower/schema.json
            image: gcr.io/ml-pipeline/ml-pipeline-kubeflow-trainer
            name: tensorflow_

**Approach one**: change master to chief
**Result**: the same error.
**Approach two**: change master to worker
**Result**: the training process got through, but the tensorflow model was not exported as we expected in the final destination(/export/export).
**Approach three**: add both chief and worker
**Result**: the training process got through, but the tensorflow model was not exported as we expect in the final destination(/export/export).

**Comment by qimingj** [comment]
To add more context: we are using tf's estimator trainer with TF version 1.6. The trainer code is like:

  ...
  exporter = tf.estimator.FinalExporter('export', serving_input_fn)
  train_spec = tf.estimator.TrainSpec(input_fn=train_input_fn, max_steps=args.steps)
  eval_spec = tf.estimator.EvalSpec(input_fn=eval_input_fn, exporters=[exporter])
  tf.estimator.train_and_evaluate(estimator, train_spec, eval_spec)
  ...

**Comment by jlewi** [comment]
This is most likely a problem in your code not the TFJob controller. The logs indicate your job is starting.
The estimator API is supposed to parse TF_CONFIG and invoke appropriate logic to checkpoint the model.

I forget at which version the estimator API changed from master to chief.

You should look at the TF.estimator API and figure out what it expects for the environment variable TF_CONFIG and then compare that against what's being set.

**Comment by jlewi** [comment]
gaoning777 reported that with TF 1.8

```
apiVersion: kubeflow.org/v1alpha2
kind: TFJob
metadata:
  name: trainer-tfjob2
spec:
  tfReplicaSpecs:
    Chief:
      replicas: 1
      restartPolicy: OnFailure
      template:
        spec:
          containers:
          - command:
            - python
            - -m
            - trainer.task
            - --optimizer=Adagrad
            - --target=label
            - --learning-rate=0.1
            - --job-dir=gs://ngao-test-mlpipeline-userguide/flowermodel/kubeflow-training-974xq/train
            - --hidden-layer-size=100,50
            - --steps=2000
            - --transformed-data-dir=gs://ngao-test-mlpipeline-userguide/flowermodel/kubeflow-training-974xq/transformed
            - --schema=gs://ml-pipeline-playground/flower/schema.json
            image: gcr.io/ml-pipeline/ml-pipeline-kubeflow-trainer
            name: tensorflow
```

Resulted in

```
The error is the same "tensorflow:Waiting for model to be ready.  Ready_for_local_init_op:  Variables not initialized:"
```

I wonder if this is because there is no worker?  Although the problems with 1.6 suggest that is a problem with not properly detecting the chief.

**Comment by jlewi** [comment]
Duplicate of kubeflow/tf-operator#761
